### PR TITLE
Update styling in mentions auto-suggestion component

### PR DESCRIPTION
**Pull request by szebniok**
### Details

I took a guess with the margins between the avatar, display name and email, as they were not specified exactly in the issue. 
@puneetlath Is the highlighted part of the email name readable enough? It wasn't shown on any of the mocks inside the design doc. 

### Fixed Issues

$ https://github.com/Expensify/App/issues/18795

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Type in <kbd>@</kbd> into the text input
2. Verify that the styles of suggestions match the mocks from https://github.com/Expensify/App/issues/18795 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

image
image

</details>

<details>
<summary>Mobile Web - Safari</summary>

image

image

</details>

<details>
<summary>Desktop</summary>
image
image

</details>

<details>
<summary>iOS</summary>

image

image

</details>

<details>
<summary>Android</summary>

image
image

</details>

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
@szebniok Looks like there is a padding on the top (and probably at the bottom) of the suggestion list in the mock.

|Mock|PR|
:----:|:-----:
Screenshot 2023-05-15 at 19 05 45 | Screenshot 2023-05-15 at 19 21 50

**Comment by szebniok** [comment]
> Hm, yeah I agree that is there on the mock, but it looks like we don't do it for emojis today. 

Yes, currently the height of the emoji and mention suggestions is exactly the same.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2023-05-15 at 20 36 30


</details>

<details>
<summary>Mobile Web - Chrome</summary>
Screenshot 2023-05-15 at 20 57 23

</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2023-05-15 at 20 55 02



</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-05-15 at 20 40 02

</details>

<details>
<summary>iOS</summary>
Screenshot 2023-05-15 at 20 50 55

</details>

<details>
<summary>Android</summary>
Screenshot 2023-05-15 at 20 51 10



</details>

**Comment by puneetlath** [comment]
@szebniok would you be able to pick this back up and take it to the finish line this week?

**Comment by szebniok** [comment]
@puneetlath I've updated the padding and the color of the higlighted text. I've updated the screenshots in the first comment.

**Comment by puneetlath** [comment]
Ok nice. I notice in your screenshots that it seems to look different on Android though.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/ec6e4535-2406-47c8-bb16-f0b2a20c1735

Screenshot 2023-05-27 at 11 03 52

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25876548/67b7a8af-9b48-4674-8661-9d1ac03d0f0f

Screenshot 2023-05-27 at 11 03 05

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/f21d7a3e-be3f-4ade-a80b-5c966a7dc851

Screenshot 2023-05-27 at 10 58 00

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/2d7a6699-da47-411b-b254-c2c56fceb4fa

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/25876548/ca6ee95e-99d0-4fee-bb4f-16efac123f34

Screenshot 2023-05-27 at 10 55 12


</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/04e6582a-aed0-4e44-9d18-07bb97e6ae7d

Screenshot 2023-05-27 at 10 59 53

</details>

**Comment by sobitneupane** [comment]
> Ok nice. I notice in your screenshots that it seems to look different on Android though.

+1

**Comment by szebniok** [comment]
Fixed the font weight on native Android, I believe that now it is consistent.

| iOS | Android |
| - | - |
| image |  image |

**Review by sobitneupane — DISMISSED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by shawnborton** [comment]
Hmm this feels like the styling is still quite a bit off. 

Each row should be 40px tall:
image

The icons that are not avatars should be green. They should be at 16x16. They should also be centered inside of a 24x24 box, this way the box matches the same size as the avatars:
image

And there should be 8px padding on the left/right side of the rows, and then 8px margin in between avatar + name + subname.

**Comment by szebniok** [comment]
@shawnborton Thank you for providing the exact pixel values. I didn't have access to the Figma. I've updated the PR and posted new screenshots in the first comment

**Comment by shawnborton** [comment]
Nice, I think this feels good to me. @puneetlath let me know what you think as well.

**Comment by sobitneupane** [comment]
@szebniok Looks like you have not pushed the commit yet. I cannot find the commit.

Screenshot 2023-05-31 at 18 11 40

**Comment by szebniok** [comment]
@sobitneupane 😅 pushed it just now.

**Comment by puneetlath** [comment]
Looks good to me too!

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-06-01 at 22 04 17
Screenshot 2023-06-01 at 22 04 25

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-06-01 at 22 15 05

</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2023-06-01 at 22 14 32


</details>

<details>
<summary>Desktop</summary>


Screenshot 2023-06-01 at 22 15 20

</details>

<details>
<summary>iOS</summary>
Screenshot 2023-06-01 at 22 15 49

</details>

<details>
<summary>Android</summary>

Screenshot 2023-06-01 at 22 21 46


</details>

**Review by sobitneupane — APPROVED** [review]

**Comment by puneetlath** [comment]
@sobitneupane just to confirm - you've tested the updated PR on all devices?

File: src/components/MentionSuggestions.js [hunk 1]
```diff
@@ -63,25 +63,42 @@ const MentionSuggestions = (props) => {
      * @returns {JSX.Element}
      */
     const renderSuggestionMenuItem = (item) => {
-        const displayedText = _.uniq([item.text, item.alternateText]).join(' - ');
-        const styledTextArray = getStyledTextArray(displayedText, props.prefix);
+        const isIcon = item.text === CONST.AUTO_COMPLETE_SUGGESTER.HERE_TEXT;
```
Comment (puneetlath) [inline src/components/MentionSuggestions.js:66]: NAB: this seems clearer to me. 

```suggestion
        const avatarSize = item.text === CONST.AUTO_COMPLETE_SUGGESTER.HERE_TEXT ? CONST.AVATAR_SIZE.MENTION_ICON : CONST.AVATAR_SIZE.SMALLER;
```

**Review by puneetlath — APPROVED** [review]
Looks good to me! 

@szebniok @sobitneupane I had one small suggestion, but I don't feel strongly, so if you disagree I'm happy to merge.

**Comment by sobitneupane** [comment]
@puneetlath Yes, I have. Should I add screenshots?

**Comment by puneetlath** [comment]
@sobitneupane yes if you could update the screenshots that would be great. Just because we saw issues with android styling before, so I want to be confident.

**Comment by sobitneupane** [comment]
@puneetlath Updated the screenshots.

**Comment by puneetlath** [comment]
Awesome, thanks!

File: src/components/MentionSuggestions.js (hunk 1 above)
Comment (puneetlath) [reply src/components/MentionSuggestions.js:66]: @szebniok thoughts? Last outstanding thing before I merge.

File: src/components/MentionSuggestions.js [hunk 2]
```diff
@@ -63,25 +63,42 @@ const MentionSuggestions = (props) => {
...
+        const styledDisplayName = getStyledTextArray(item.text, props.prefix);
+        const styledHandle = getStyledTextArray(item.alternateText, props.prefix);
```
Comment (narefyev91) [inline src/components/MentionSuggestions.js:68]: https://github.com/Expensify/App/pull/21779/files - bug two same emails display for user without display name

File: src/styles/StyleUtils.js [hunk 3]
```diff
@@ -1028,13 +1029,15 @@ function getAutoCompleteSuggestionItemStyle(highlightedEmojiIndex, rowHeight, ho
 function getAutoCompleteSuggestionContainerStyle(itemsHeight, shouldIncludeReportRecipientLocalTimeHeight) {
     const optionalPadding = shouldIncludeReportRecipientLocalTimeHeight ? CONST.RECIPIENT_LOCAL_TIME_HEIGHT : 0;
     const padding = CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTER_PADDING - optionalPadding;
+    const borderWidth = 2;
+    const height = itemsHeight + 2 * CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTER_INNER_PADDING + borderWidth;
```
Comment (cubuspl42) [inline src/styles/StyleUtils.js:1033]: Coming from https://github.com/Expensify/App/issues/27140

When changing this logic, we didn't preserve the fix introduced in https://github.com/Expensify/App/issues/18112

**Pull request merged**